app/src/main/java/com/rockthevote/grommet/ui/registration/name/PersonName.kt

```diff
@@ -0,0 +1,56 @@
+package com.rockthevote.grommet.ui.registration.name
+
+import android.text.Editable
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.EditText
+import com.rockthevote.grommet.R
+import com.rockthevote.grommet.data.db.model.Prefix
+import com.rockthevote.grommet.data.db.model.Suffix
+import com.rockthevote.grommet.databinding.ViewNameBinding
+import com.rockthevote.grommet.ui.views.NameView
+
+data class PersonName(
+    // Required Fields
+    val firstName: String,
+    val title: Prefix,
+    val lastName: String,
+    // Optional
+    val middleName: String?,
+    val suffix: Suffix?
+)
+
+@Throws(IllegalArgumentException::class)
+fun NameView.toPersonName(): PersonName? {
```

Comment:
is this the only data class that also does data binding? Looks like it breaks the pattern you were going for (comparing it to the `Address.kt` and then `PersonalInfoData.kt` and` PersonalInfoExt.kt`).

Reply:
Guessing it might not have been very straight forward given how some of the views are componentized and re-used and other view data is just done "in-place" in the fragment?